snekmer/scripts/learn_learn.py

```diff
@@ -0,0 +1,235 @@
+# ---------------------------------------------------------
+# Imports
+# ---------------------------------------------------------
+
+import pickle
+from datetime import datetime
+from os import makedirs
+from os.path import exists, join
+
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import snekmer as skm
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split, StratifiedKFold
+from sklearn.preprocessing import LabelEncoder
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import pyarrow.csv as csv
+import seaborn as sns
+import sklearn
+from Bio import SeqIO
+from scipy.interpolate import interp1d
+from scipy.stats import rankdata
+import random
+import snekmer as skm
+from scipy.ndimage import gaussian_filter1d
+import sklearn.metrics.pairwise
+import itertools
+import os
+import shutil
+import sys
+import re
+# ---------------------------------------------------------
+# Files and Parameters
+# ---------------------------------------------------------
+
+config = snakemake.config
+
+# change matplotlib backend to non-interactive
+plt.switch_backend("Agg")
+
+outDir = skm.io.define_output_dir(
+    config["alphabet"], config["k"], nested=config["nested_output"]
+)
+
+# ---------------------------------------------------------
+# Run script
+# ---------------------------------------------------------
+
+class Library:
+    """
+Initializes a Library object.
+
+This object is responsible for processing annotations and kmer data.
+It creates a kmer-count matrix for each input file.
+
+Attributes:
+annotation (list): A list to store annotations loaded from files.
+seqAnnot (dict): A dictionary mapping sequence IDs to their annotations.
+kmerList (list): A list of unique kmers present in the data.
+df (DataFrame or None): DataFrame containing kmer data.
+seqids (list): A list of sequence IDs.
+kmerTotals (list): A list to store total counts of each k-mer across all sequences.
+seqKmerdict (dict): A dictionary mapping sequence IDs to their k-mer counts.
+annotationCounts (dict): A dictionary mapping annotations to their counts.
+totalSeqs (int): The total number of sequences after filtering.
+"""
+
+    def __init__(self):
+        self.annotation = []
+        self.seqAnnot = {}
+        self.kmerList = []
+        self.df = None
+        self.seqids = []
+        self.kmerTotals = []
+        self.seqKmerdict = {}
+        self.annotationCounts = {}
+        self.totalSeqs = 0
+
+    def loadAnnotations(self, input_annotation):
+        """
+Load annotations from a list of provided input files.
+
+Args:
+    input_annotation (list): List of file paths containing annotations.
+"""
+        for f in input_annotation:
```

Comment:
Please see previous comments about consistency in variable naming and apply to this entire file